tests/model_notebooks/test_notebooks.py

```diff
@@ -0,0 +1,361 @@
+            raise FileNotFoundError(
+                f"Notebook directory does not exist: {self.notebook_directory}"
+            )
+
+        if not self.notebook_directory.is_dir():
+            raise NotADirectoryError(f"Path is not a directory: {self.notebook_directory}")
+
+        # Ensure the required Jupyter kernel is registered
+        self._ensure_kernel_registered()
+
+    def _check_kernel_exists(self) -> bool:
+        """
+        Check if the required Jupyter kernel is already registered.
+
+        Returns:
+            True if kernel exists, False otherwise
+        """
+        try:
+            result = subprocess.run(
+                ["jupyter", "kernelspec", "list", "--json"],
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+            kernelspecs = json.loads(result.stdout)
+            kernel_exists = self.kernel_name in kernelspecs.get("kernelspecs", {})
+
+            if kernel_exists:
+                logger.debug(f"Kernel '{self.kernel_name}' is already registered")
+            else:
+                logger.debug(f"Kernel '{self.kernel_name}' not found")
+
+            return kernel_exists
+        except (subprocess.CalledProcessError, json.JSONDecodeError, FileNotFoundError) as e:
+            logger.warning(f"Could not check kernel specs: {e}")
+            return False
+
+    def _register_kernel(self) -> bool:
+        """
+        Register the conda environment as a Jupyter kernel.
+
+        Returns:
+            True if registration succeeded, False otherwise
+        """
+        try:
+            logger.info(
+                f"Registering kernel '{self.kernel_name}' for conda env '{self.conda_env_name}'"
+            )
+
+            # Use conda run to execute ipykernel install in the target environment
+            result = subprocess.run(
+                [
+                    "conda",
+                    "run",
+                    "-n",
+                    self.conda_env_name,
+                    "python",
+                    "-m",
+                    "ipykernel",
+                    "install",
+                    "--user",
+                    "--name",
+                    self.kernel_name,
+                    "--display-name",
+                    f"Python (vivarium_gates_mncnh_{self.environment_type})",
+                ],
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+
+            logger.success(f"Successfully registered kernel '{self.kernel_name}'")
+            logger.debug(f"Output: {result.stdout}")
+            return True
+
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Failed to register kernel: {e.stderr}")
+            return False
+        except FileNotFoundError:
+            logger.error("conda command not found. Ensure conda is in PATH.")
+            return False
+
+    def _ensure_kernel_registered(self) -> None:
```

Comment:
For reasons I do not fully understand, you cannot just pass a kernel name into the pm.execute_notebook() call with slurm, you have to make sure the kernel (or conda envrionment) is registered on slurm node.